Source/Xamarin/Prism.Forms/Navigation/Xaml/Navigation.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Windows.Input;
+using Prism.Common;
+using Prism.Ioc;
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace Prism.Navigation.Xaml
+{
+    public abstract class Navigation : IMarkupExtension, ICommand
```

Comment:
let's split this up to a static Navigation class with the BindableProperties and an abstract NavigationExtension class that implements IMarkupExtension and ICommand